### PR TITLE
Add support for OpenShift DeploymentConfig

### PR DESCRIPTION
**Pull request by hossainemruz**
Fix: https://github.com/appscode/stash/issues/161

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -102,3 +102,5 @@ import:
   version: master
 - package: kmodules.xyz/custom-resources
   version: release-10.0
+- package: kmodules.xyz/openshift
+  version: master
```
Comment (tamalsaha) [inline glide.yaml]: version: release-10.0

File: hack/docker/setup.sh [hunk 2]
```diff
@@ -82,6 +82,9 @@ build_docker() {
   cat >Dockerfile <<EOL
 FROM alpine:3.8
 
+# add our user and group first to make sure their IDs get assigned consistently, regardless of whatever dependencies get added
+RUN addgroup -S stash && adduser -S -G stash stash
```
Comment (tamalsaha) [inline hack/docker/setup.sh]: We should use a number as the user id instead of username.

File: glide.yaml (hunk 1 above)
Comment (hossainemruz) [reply glide.yaml]: fixed

File: hack/docker/setup.sh (hunk 2 above)
Comment (hossainemruz) [reply hack/docker/setup.sh]: Used numeric id

File: pkg/controller/backup_session.go [hunk 3]
```diff
@@ -82,15 +82,21 @@ func (c *StashController) runBackupSessionProcessor(key string) error {
...
 	glog.Infof("Sync/Add/Update for BackupSession %s", backupSession.GetName())
 
+	if backupSession.Status.Phase == api_v1beta1.BackupSessionFailed ||
+		backupSession.Status.Phase == api_v1beta1.BackupSessionSucceeded {
+		log.Infof("Skipping processing BackupSession %s/%s. Reason: phase is %q.", backupSession.Namespace, backupSession.Name, backupSession.Status.Phase)
+		return nil
+	}
+
 	// check weather backup session is completed or running and set it's phase accordingly
 	phase, err := c.getBackupSessionPhase(backupSession)
```
Comment (tamalsaha) [inline pkg/controller/backup_session.go]: What happens to this error?

Comment (hossainemruz) [reply pkg/controller/backup_session.go]: It's not error of operator itself. It is the error that failed backup for a host that we wrote in `backupOutput`. So, we just used this error to write failure event.

**Pull request merged**